doc/contributing/node-api-tooling.md

```diff
@@ -0,0 +1,179 @@
+
+- Updating Markdown `code` blocks by adding Language highlighting
+  - It also adds the "CJS"/"MJS" switch to Nodes that are followed by their CJS/ESM equivalents.
+- Increasing the Heading level of each Heading
+- Parses YAML blocks and transforms them into HTML elements (See more at the `parseYAML` method)
+- Updates BlockQuotes that are prefixed by the "Stability" word into a Stability Index HTML element.
+
+#### `parseYAML`
+
+**New Tooling:** Most of the features within this method are available within the new tooling.
+
+This method is responsible for parsing the `<--YAML snippets -->` and transforming them into HTML elements.
+
+It follows a certain kind of "schema" that basically constitues in the following options:
+
+| YAML Key | Description | Example | Example Result | Available on new tooling |
+| -------- | ----------- | ------- | -------------- | ------------------------ |
+| `added`  | It's used to reference when a certain "module", "class" or "method" was added on Node.js | `added: v0.1.90` | `Added in: v0.1.90` | Yes |
+| `deprecated` | It's used to reference when a certain "module", "class" or "method" was deprecated on Node.js | `deprecated: v0.1.90` | `Deprecated since: v0.1.90` | Yes |
+| `removed` | It's used to reference when a certain "module", "class" or "method" was removed on Node.js | `removed: v0.1.90` | `Removed in: v0.1.90` | No |
+| `changes` | It's used to describe all the changes (historical ones) that happened within a certain "module", "class" or "method" in Node.js | `[{ version: v0.1.90, pr-url: '', description: '' }]` | -- | Yes |
+| `napiVersion` | It's used to describe in which version of the N-API this "module", "class" or "method" is available within Node.js | `napiVersion: 1` | `N-API version: 1` | Yes |
+
+**Note.:** The `changes` field gets preprended with the `added`, `deprecated` and `removed` fields if they exist. The table only gets generated if a `changes` field exists. In the new tooling only "added" is prepended for now.
+
+#### `buildToc`
+
+**New Tooling:** This feature is natively available within the new tooling through MDX.
+
+This method generates the Table of Contents based on all the Headings of the Markdown file.
+
+#### `altDocs`
+
+**New Tooling:** All features within this method are available within the new tooling.
+
+This method generates a version picker for the current page to be shown in older versions of the API docs.
+
+### `json.mjs`
+
+This file is responsible for generating a JSON object that (supposedly) is used for IDE-Intellisense or for indexing of all the "methods", "classes", "modules", "events", "constants" and "globals" available within a certain Markdown file.
+
+It attempts a best effort extraction of the data by using several regular expression patterns (ReGeX).
+
+**Note.:** JSON output generation is currently not supported by the new tooling, but it is in the pipeline for development.
+
+#### `jsonAPI`
```

Comment:
A number of the headings are missing the prefix like `json.` in this case. Others have them. To be consistent we should include the prefix in all cases.

Reply:
Hmm, re-reading this comment, could you rephrase it? I'm not sure I got it 🙈

Reply:
Re-reading I think I missed the levesl of the headers and its ok, so never mind.